.github/workflows/build-pr.yml

```diff
@@ -189,8 +194,28 @@ jobs:
         if: ${{ always() }}
         uses: actions/upload-artifact@v4
         with:
-          name: docker-logs-jdk${{ matrix.jdk }}
+          name: docker-logs-${{ github.job }}-jdk${{ matrix.jdk }}
           path: |
             ${{ env.docker_log_file_oss }}
             ${{ env.docker_log_file_ee }}
 
+  test-push:
+    name: Test pushing image (dry run)
+    needs: [ prepare ]
+    uses: ./.github/workflows/tag_image_push.yml
+    with:
+      HZ_VERSION: ${{ needs.prepare.outputs.LAST_RELEASED_HZ_VERSION_OSS }}
+      RELEASE_VERSION: ${{ needs.prepare.outputs.LAST_RELEASED_HZ_VERSION_OSS }}
+      RELEASE_TYPE: ALL
+      DRY_RUN: true
+    secrets: inherit
+
+  test-push-rhel:
+    name: Test pushing image RHEL (dry run)
+    needs: [ prepare ]
+    uses: ./.github/workflows/tag_image_push_rhel.yml
+    with:
+      HZ_VERSION: ${{ needs.prepare.outputs.LAST_RELEASED_HZ_VERSION_OSS }}
+      RELEASE_VERSION: ${{ needs.prepare.outputs.LAST_RELEASED_HZ_VERSION_OSS }}
+      DRY_RUN: true
+    secrets: inherit
```

Comment:
I'm concerned that we pass all the secrets needed to do the actual push to the prod registry. What if one creates a PR and modifies `tag_image_push.yml` that DRY_RUN is not working anymore? We would end up with pushing broken image to prod. 

WDYT about passing here secrets explicitly and only the ones that are used for read-only operations?

Reply:
> WDYT about passing here secrets explicitly and only the ones that are used for read-only operations?

I was trying to avoid a change to the `push` action necessitating a change to `build-pr` as well, but I can see your concern. 

However, if they are passed explicitly, how are they accessed in the workflow? AFAIK with the `with` syntax it passes them as environment variables, so the `${{ secrets.MY_CRED }}" would fail if `MY_CRED` was instead an environment variable?

Reply:
There's also `secrets` section. Check the first example https://docs.github.com/en/enterprise-cloud@latest/actions/sharing-automations/reusing-workflows#passing-inputs-and-secrets-to-a-reusable-workflow: 
```
jobs:
  call-workflow-passing-data:
    uses: octo-org/example-repo/.github/workflows/reusable-workflow.yml@main
    with:
      config-path: .github/labeler.yml
    secrets:
      personal_access_token: ${{ secrets.token }}
```

Reply:
I've tried in f1f1a3f60753fd6e1df7a94e7656d2d5af27d1cd

However the RHEL `push` needs the credentials for pushing, but _also_ the preflight etc stuff we actually want to test, so it's still `inherit`.